packages/codemods/src/plugins/contract.ts

```diff
@@ -0,0 +1,62 @@
+import { ImportDeclaration, Collection, JSCodeshift } from 'jscodeshift';
+
+// We want to ensure the config contract is correct so devs can get type safety
+type ValidateConfig<T, Struct> = T extends Struct
+  ? Exclude<keyof T, keyof Struct> extends never
+    ? T
+    : never
+  : never;
+
+type BaseConfig = { j: JSCodeshift };
+
+export interface CodemodPlugin {
```

Comment:
Later on this interface might not scale, as it's specific to changing away from one library to compiled (i.e. the codemods currently available). It should be ok for now though.

Reply:
I can rename this to be specific and export it non-default to make the transition later easier